Finalização de serviços de usuário e de caronas

rornellas opened the pull request.

Commit:
Serviço 80%

Commit:
Mudança no nome do readme

rornellas committed:
Update README.md

Commit:
Mudança no nome do readme

Commit:
Merge remote-tracking branch 'origin/servicos_user_ride' into servicos_user_ride

Commit:
Ajuste na exibição do hateoas, mudança para servlets do webffux, ajuste nova versão do swagger (3.0.0), metodo patch do controller rides agora retorna ResponseEntity<Void> para entrar em conformidade com o HATEOAS

Commit:
Imports itunuzadis

Commit:
Imports itunuzadis

Commit:
Inclusão de acesso ao mongodb por dns pelo serviço

WagnerCarvalho commented on src/main/kotlin/br/com/fiap/carsocial/service/api/controller/response/RideResponse.kt:
```diff
@@ -0,0 +1,21 @@
+        var email: String? = null,
+        var car: Car?=null,
+        var coords: CoordsResponse?=null
+) {
```
Você precisa utilizar constructor em kotlin, pois já está iniciando as variáveis

rornellas replied:
É um data class, então esse outro construtor é opcional. Serve para construir um build response a partir de um Ride (document do mongodb). Recomenda outra forma de fazer isso?

WagnerCarvalho replied:
Pode passar o Objeto inteiro se é Igual, me parece que vc passa Ride para Response...
Menos código repetido!!!

`
data class RideResponse(
    var results: Ride = Ride()
)
override fun create(userRequest: RideRequest): Mono<RideResponse> {
        return rideRepository.save(userRequest.convertToRide())
            .map {
                RideResponse(it)
         }
}
`

rornellas replied:
Entendi... mas isso não seria perigoso.. tipo se no futuro fosse adicionada alguma informação que não quero passar no response ao objeto Ride... se alguém esquecesse disso poderia ser um problema. Eu estaria passando um documento do mongo diretamente para o response. O que acha?

WagnerCarvalho commented on src/main/resources/application-local.properties:
```diff
@@ -0,0 +1 @@
+spring.data.mongodb.host=192.168.99.100
```
O que seria esse IP ?

rornellas replied:
Esse ip é o que uso localmente. Quando comitamos, é uma má pratica subir os outros profiles?

WagnerCarvalho commented on src/test/kotlin/br/com/fiap/carsocial/service/ServiceApplicationTests.kt:
```diff
@@ -1,13 +1,10 @@
-	}
-
-}
+//@SpringBootTest
```
deletar o arquivo!

rornellas replied:
kkk, ok

WagnerCarvalho commented on Dockerfile:
```diff
@@ -0,0 +1,14 @@
+USER root
+RUN  apk update && apk upgrade && apk add netcat-openbsd
+RUN mkdir -p /usr/local/service
+ADD build/libs/service-0.0.1-SNAPSHOT.jar /usr/local/service-app/service-0.0.1-SNAPSHOT.jar
```
Precisa adicionar na documentação que antes de executar o Docker deverá gerar o binário da aplicação... 
Pq quando o usuário baixa o código do github não terá o binário **"service-0.0.1-SNAPSHOT.jar"**.. 
Para isso precisa fazer um antes de executar o docker ./gradlew bootJar no diretório raiz do projeto.

WagnerCarvalho commented on README.md:
```diff
@@ -0,0 +1,9 @@
+Para executar o aplicativo, é necessário ter o docker instalado.
+
+#EXECUTAR
```
Adiconar o ponto acima...
exemplo:
cd ${para raiz do projeto} && ./gradlew bootJar && cd ..

WagnerCarvalho commented on src/main/kotlin/br/com/fiap/carsocial/service/ServiceApplication.kt:
```diff
@@ -3,8 +3,11 @@ package br.com.fiap.carsocial.service
+
 @SpringBootApplication
-class ServiceApplication
+class ServiceApplication {
```
é necessário ter isso o pode remover ?

rornellas replied:
Removi já

WagnerCarvalho commented on src/main/kotlin/br/com/fiap/carsocial/service/api/controller/request/CoordsRequest.kt:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.data.mongodb.core.geo.GeoJsonPoint
+
+data class CoordsRequest (
+    var latitude: String?=null,
```
Colocar validação field:NotBlank ???

rornellas replied:
Então... não coloquei validação pq não vai ser obrigatório. O usuário vai se cadastrar e no ato do cadastro pode ir a geolocalização ou não. O que acha? Acha que deveria ser obrigatório?

WagnerCarvalho reviewed (commented):
Como vamos fazer com os motorista ?
    Solução 1 - Vamos carregar um base em produção com alguns motorista ? Assim quando vir uma requisição do usuário do dispositivo sempre terá motorista para renderizar no mapa.
    2 - Quando receber a coordenada do usuário do dispositivo simular motorista em raio de 5k

Adicionar no Modelo do usuário a variável telefone

rornellas commented on src/main/kotlin/br/com/fiap/carsocial/service/WebFluxConfig.kt:
```diff
@@ -0,0 +1,10 @@
+//package br.com.fiap.carsocial.service
+//
+//import org.springframework.context.annotation.Configuration
```
Opa... ok

Commit:
Ajustes de solicitações do pull request
Mudança no endereço default do diretório de dados linkado ao mongodb do docker-compose.yml

WagnerCarvalho commented on src/main/kotlin/br/com/fiap/carsocial/service/api/document/User.kt:
```diff
@@ -1,11 +1,12 @@
 data class User(
     @Id var id: String? =null,
-    var name: String?=null
+    var name: String?=null,
```
incluir telefone

rornellas replied:
Feito

Commit:
Adição do número de telefone

Commit:
Ajuste erro escrita docker-compose.yml mongo

Commit:
Ajuste erro escrita docker-compose.yml mongo

Commit:
Ajuste de cadastro do telefone

Commit:
Inclusão do jar - o ec2 não está conseguindo buildar

Commit:
adicionando nginx

Commit:
Ajuste no calculo da distância

Commit:
Comitando binário para uso em ec2

Commit:
Ajuste na atualização das coordenadas de usuário

Commit:
Distance em metros

Commit:
Distance em metros

Commit:
serviço de viagens pronto e binário para aws

Commit:
Criação de vídeo para exibição da navegação
